service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -109,6 +121,55 @@ public void preparePipelineRun(
 
     // increment the prepare metric for this pipeline
     MetricsUtils.incrementPipelinePrepareRun(pipelineName);
+
+    return pipelineFileInputs;
+  }
+
+  /**
+   * Generate signed PUT urls and curl commands for each user-provided file input in the pipeline.
+   *
+   * <p>Each user-provided file input (assumed to be a path to a local file) is translated into a
+   * write-only (PUT) signed url in a location in the pipeline workspace storage container, in a
+   * directory defined by the jobId.
+   *
+   * <p>This signed url along with the source file path provided by the user are used to generate a
+   * curl command that the user can run to upload the file to the location in the pipeline workspace
+   * storage container.
+   */
+  private Map<String, Map<String, String>> prepareFileInputs(
+      Pipeline pipeline, UUID jobId, Map<String, Object> userProvidedInputs) {
+    // get the list of files that the user needs to upload
+    List<String> fileInputNames =
+        pipeline.getPipelineInputDefinitions().stream()
+            .filter(PipelineInputDefinition::getUserProvided)
+            .filter(p -> p.getType().equals(PipelineVariableTypesEnum.FILE))
+            .map(PipelineInputDefinition::getName)
+            .toList();
+
+    // TODO don't hardcode this
+    String projectId = "terra-dev-244bacba";
+    String bucketName = pipeline.getWorkspaceStorageContainerUrl().replace("gs://", "");
```

Comment:
should we remove the `gs://` prefix here, or should we store the bucket name without the `gs://`?

Reply:
is we're calling it `Url` then it shoudl have gs://.  we could call it just `Name` instead and then use it appropriately in other places

Reply:
i like that change - added that to AC for TSPS-307

Reply:
if https://github.com/DataBiosphere/terra-scientific-pipelines-service/pull/118 is merged first, will need to update this